src/v/cluster/scheduling/allocation_node.cc

```diff
@@ -73,13 +77,24 @@ void allocation_node::deallocate_on(
     allocation_capacity& domain_partitions
       = _allocated_domain_partitions[domain];
     vassert(
-      domain_partitions > allocation_capacity{0}
-        && domain_partitions <= _allocated_partitions,
-      "Unable to deallocate partition from core {} in domain {} at node {}",
-      core,
+      domain_partitions <= _allocated_partitions,
```

Comment:
Can't this remaining conditin still fail in the same way? E.g., domain 0 has 0 allocated partions and domain 1 has 1. 

So we have { 0, 1 } for domain allocated and 1 global allocated.

Then a dealloc request comes in for domain 1: with relaxed policy, we will now skip the --domain_partitions but we will --allocated_partitions. So now we (still) have {0, 1} domain allocated but 0 global and this assert will trigger when we start processing domain 1?

Reply:
I'm thinking of that since I've posted this PR. I think a much better way to go here would be this:
* In case of the `!(domain_partitions > allocation_capacity{0})`, do not change `allocation_node` state at all. The most probable reason for this condition is a repeated delete topic operation. While trying to tolerate a repeated deletion, it's not a good idea to update either `weights` or `allocated_partitions`.
* Do not try to relax any other invariants yet before we observe a case when relaxing them can help in a specific case.